stores/sql/mysql/main.go

```diff
@@ -25,64 +27,70 @@ func NewMainDatabase(db *dsql.DB, log *zap.SugaredLogger, lqd, ltd time.Duration
 	}
 }
 
+func (b *MainDatabase) ApplyMigration(fn func(tx sql.Tx) (bool, error)) error {
+	return applyMigration(b.db, fn)
+}
+
 func (b *MainDatabase) Close() error {
 	return b.db.Close()
 }
 
+func (b *MainDatabase) DB() *sql.DB {
+	return b.db
+}
+
+func (b *MainDatabase) CreateMigrationTable() error {
+	return createMigrationTable(b.db)
+}
+
+func (b *MainDatabase) MakeDirsForPath(tx sql.Tx, path string) (uint, error) {
+	insertDirStmt, err := tx.Prepare("INSERT INTO directories (name, db_parent_id) VALUES (?, ?) ON DUPLICATE KEY UPDATE id = id")
```

Comment:
you can use `RETURN ID` here to save the `queryDirStmt`

Reply:
I think `RETURN` doesn't work like `RETURNING`? In MySQL it can only be used in stored procedures I afaik.

Reply:
MySQL has https://dev.mysql.com/doc/refman/8.0/en/information-functions.html#function_last-insert-id